internal/controller/eventing/integrationtests/controller/integration_test.go

```diff
@@ -604,6 +438,246 @@ func Test_WatcherEventingCRK8sObjects(t *testing.T) {
+				utils.WithEventMeshBackend("test-secret-name2"),
+				utils.WithEventingPublisherData(2, 2, "199m", "99Mi", "399m", "199Mi"),
+				utils.WithEventingEventTypePrefix("test-prefix"),
+			),
+			givenDeploymentReady: true,
+			wantMatches: gomega.And(
+				matchers.HaveStatusReady(),
+				matchers.HaveEventMeshSubManagerReadyCondition(),
+				matchers.HavePublisherProxyReadyConditionDeployed(),
+				matchers.HaveFinalizer(),
+			),
+			wantEnsureK8sObjects: true,
+		},
+		{
+			name: "Eventing CR should have processing state when deployment is not ready yet",
+			givenEventing: utils.NewEventingCR(
+				utils.WithEventMeshBackend("test-secret-name3"),
+				utils.WithEventingPublisherData(2, 2, "199m", "99Mi", "399m", "199Mi"),
+				utils.WithEventingEventTypePrefix("test-prefix"),
+			),
+			wantMatches: gomega.And(
+				matchers.HaveStatusProcessing(),
+				matchers.HaveEventMeshSubManagerReadyCondition(),
+				matchers.HavePublisherProxyReadyConditionProcessing(),
+				matchers.HaveFinalizer(),
+			),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+
+			// given
+			eventingcontroller.IsDeploymentReady = func(deployment *v1.Deployment) bool {
+				return tc.givenDeploymentReady
+			}
+
+			// create unique namespace for this test run.
+			givenNamespace := tc.givenEventing.Namespace
+
+			testEnvironment.EnsureNamespaceCreation(t, givenNamespace)
+
+			// create eventing-webhook-auth secret.
+			testEnvironment.EnsureOAuthSecretCreated(t, tc.givenEventing)
+
+			if !tc.shouldFailSubManager {
+				// create EventMesh secret.
+				testEnvironment.EnsureEventMeshSecretCreated(t, tc.givenEventing)
+			}
+
+			// when
+			// create Eventing CR.
+			testEnvironment.EnsureK8sResourceCreated(t, tc.givenEventing)
+
+			defer func() {
+				testEnvironment.EnsureEventingResourceDeletion(t, tc.givenEventing.Name, givenNamespace)
+				if !*testEnvironment.EnvTestInstance.UseExistingCluster && !tc.shouldFailSubManager {
+					testEnvironment.EnsureDeploymentDeletion(t, eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
+				}
+			}()
+
+			// then
+			// check Eventing CR status.
+			testEnvironment.GetEventingAssert(g, tc.givenEventing).Should(tc.wantMatches)
+			if tc.givenDeploymentReady {
+				// check if EPP deployment, HPA resources created and values are reflected including owner reference.
+				ensureEPPDeploymentAndHPAResources(t, tc.givenEventing, testEnvironment)
+				// TODO: ensure NATS Backend config is reflected. Done as subscription controller is implemented.
+			}
+
+			if tc.wantEnsureK8sObjects {
+				// check if other EPP resources exists and values are reflected.
+				ensureK8sResources(t, tc.givenEventing, testEnvironment)
+				// check if webhook configurations are updated with correct CABundle.
+				testEnvironment.EnsureCABundleInjectedIntoWebhooks(t)
+			}
+		})
+	}
+}
+
+// This test should run after Test_CreateEventingCR_EventMesh
```

Comment:
Why the tests needs to be in sequential order? All the tests should be independent of each other and should be able to run in parallel.

Reply:
https://github.com/kyma-project/eventing-manager/pull/72/files#r1311277763